File: Leetcode/0168_excel_sheet_column_title/0168_excel_sheet_column_title.go
package excelsheetcolumntitle

func convertToTitle(columnNumber int) string {
    alpha := "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    var result string

    for columnNumber > 0 {
        if columnNumber%26 == 0 {
            result += "Z"
            columnNumber = columnNumber/26 - 1
        } else {
            result += string(alpha[(columnNumber%26)-1])
            columnNumber /= 26
        }
    }

    return reverseString(result)
}

// Function to reverse a string
func reverseString(s string) string {
    runes := []rune(s)
    for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
        runes[i], runes[j] = runes[j], runes[i]
    }
    return string(runes)
}

File: Leetcode_past/algorithms/easy/0168_excel_sheet_column_title/0168_excel_sheet_column_title.cs
public class Solution {
    public string ConvertToTitle(int columnNumber) {
        string alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        string result = "";

        while (columnNumber > 0) {
            if (columnNumber % 26 == 0) {
                result += "Z";
                columnNumber = columnNumber / 26 - 1;
            } else {
                result += alpha[(columnNumber % 26) - 1];
                columnNumber /= 26;
            }
        }

        return ReverseString(result);
    }

    // Function to reverse a string
    private string ReverseString(string s) {
        char[] charArray = s.ToCharArray();
        int start = 0;
        int end = s.Length - 1;

        while (start < end) {
            char temp = charArray[start];
            charArray[start] = charArray[end];
            charArray[end] = temp;
            start++;
            end--;
        }

        return new string(charArray);
    }
}
